cfme/base/ssui.py

```diff
@@ -100,16 +100,23 @@ def login(self, user=None, **kwargs):
         password = conf.credentials['default']['password']
         cred = Credential(principal=username, secret=password)
         user = User(credential=cred)
-    login_view = navigate_to(self.appliance.server, 'LoginScreen')
-    login_view.fill({
-        'username': user.credential.principal,
-        'password': user.credential.secret,
-    })
-    login_view.login.click()
-    # Without this the login screen just exits after logging in
-    time.sleep(3)
-    login_view.flash.assert_no_error()
-    self.browser.plugin.ensure_page_safe()
+
+    logged_in_view = self.appliance.browser.create_view(SSUIBaseLoggedInPage)
+
+    if not logged_in_view.logged_in_as_user(user):
```

Comment:
Consider changing this to
```python
if logged_in_view.logged_in_as_user(user):
    return
...
```
to make it more flat.

Reply:
Then it will be like :
if logged_in_view.logged_in_as_user(user):
    return
else: 
    the whole login code

Reply:
After ``return`` there's no need for ``else``. That's the point of "return early". See e.g. http://blog.timoxley.com/post/47041269194/avoid-else-return-early
```python
if logged_in_view.logged_in_as_user(user):
    return
rest of the code
```